### PR TITLE
Fill only requested columns when querying system.parts & system.parts…

### LINKED ISSUE #19570: system.parts (and other system tables): fill only requested columns
When user requested single column from the table we still calculating all of them. Calculation of some columns is quite expensive (need to do stat calls / read files etc).

### PR DESCRIPTION
**Pull request by anmolarora**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Fill only requested columns when querying system.parts & system.parts_columns. Closes #19570.
...

### CODE REVIEW

File: src/Storages/System/StorageSystemPartsBase.cpp [hunk 1]
```diff
@@ -245,16 +245,29 @@ Pipe StorageSystemPartsBase::read(
...
+        }
+    }
+    MutableColumns res_columns = header.cloneEmptyColumns();
     if (has_state_column)
         res_columns.push_back(ColumnString::create());
 
     while (StoragesInfo info = stream.next())
     {
-        processNextStorage(res_columns, info, has_state_column);
+        processNextStorage(res_columns, std::move(columns_mask), info, has_state_column);
```
Comment (Avogar) [inline src/Storages/System/StorageSystemPartsBase.cpp:268]: You pass column_mask in processNextStorage in the loop, you can't move it.

Comment (anmolarora) [reply src/Storages/System/StorageSystemPartsBase.cpp:268]: Yes, that was an oversight. I'll fix it.

Comment (anmolarora) [reply src/Storages/System/StorageSystemPartsBase.cpp]: I've fixed this but I think now flakky tests are failing.

**Comment by Avogar** [comment]
Failed test is not related.

**Pull request merged**